src/newton_ac_powerflow.jl

```diff
@@ -0,0 +1,336 @@
+            converged, x = _newton_powerflow(pf, data; nlsolve_kwargs...)
+            if converged
+                if _check_q_limit_bounds!(data, x)
+                    return converged, x
+                end
+            else
+                return converged, x
+            end
+        end
+    else
+        return _newton_powerflow(pf, data; nlsolve_kwargs...)
+    end
+end
+
+function _newton_powerflow(
+    pf::ACPowerFlow{NLSolveACPowerFlow},
+    data::ACPowerFlowData;
+    nlsolve_kwargs...,
+)
+    pf = PolarPowerFlow(data)
+    J = PowerFlows.PolarPowerFlowJacobian(data, pf.x0)
+
+    df = NLsolve.OnceDifferentiable(pf, J, pf.x0, pf.residual, J.Jv)
+    res = NLsolve.nlsolve(df, pf.x0; nlsolve_kwargs...)
+    if !res.f_converged
+        @error(
+            "The powerflow solver NLSolve did not converge (returned convergence = $(res.f_converged))"
+        )
+    end
+    return res.f_converged, res.zero
+end
+
+function _newton_powerflow(
+    pf::ACPowerFlow{KLUACPowerFlow},
+    data::ACPowerFlowData;
+    nlsolve_kwargs...,
+)
+    # Fetch maxIter and tol from kwargs, or use defaults if not provided
+    maxIter = get(nlsolve_kwargs, :maxIter, DEFAULT_NR_MAX_ITER)
+    tol = get(nlsolve_kwargs, :tol, DEFAULT_NR_TOL)
+    i = 0
+
+    # Find indices for each bus type
+    ref = findall(x -> x == PowerSystems.ACBusTypesModule.ACBusTypes.REF, data.bus_type)
+    pv = findall(x -> x == PowerSystems.ACBusTypesModule.ACBusTypes.PV, data.bus_type)
+    pq = findall(x -> x == PowerSystems.ACBusTypesModule.ACBusTypes.PQ, data.bus_type)
+    pvpq = [pv; pq]
+
+    #nref = length(ref)
+    npv = length(pv)
+    npq = length(pq)
+    npvpq = npv + npq
+    n_buses = length(data.bus_type)
+
+    Vm = data.bus_magnitude[:]
+    # prevent unfeasible starting values for Vm; for pv and ref buses we cannot do this:
+    @. Vm[pq] = clamp.(Vm[pq], 0.9, 1.1)
+    Va = data.bus_angles[:]
+    V = zeros(Complex{Float64}, length(Vm))
+    @. V = Vm .* exp.(1im * Va)
+
+    Va_pv = view(Va, pv)
+    Va_pq = view(Va, pq)
+    Vm_pq = view(Vm, pq)
+
+    # pre-allocate dx
+    dx = zeros(Float64, npv + 2 * npq)
+
+    dx_Va_pv = view(dx, 1:npv)
+    dx_Va_pq = view(dx, (npv + 1):(npv + npq))
+    dx_Vm_pq = view(dx, (npv + npq + 1):(npv + 2 * npq))
+
+    Ybus = data.power_network_matrix.data
+
+    Sbus =
+        data.bus_activepower_injection[:] - data.bus_activepower_withdrawals[:] +
+        1im * (data.bus_reactivepower_injection[:] - data.bus_reactivepower_withdrawals[:])
+    
+    # Pre-allocate mis and F and create views for the respective real and imaginary sections of the arrays:
+    mis = zeros(Complex{Float64}, length(V))
+    mis_pvpq = view(mis, pvpq)
+    mis_pq = view(mis, pq)
+
+    F = zeros(Float64, npvpq + npq)
+    F_real = view(F, 1:npvpq)
+    F_imag = view(F, npvpq + 1:npvpq + npq)
+
+    mis .= V .* conj.(Ybus * V) .- Sbus
+    @. F_real = real(mis_pvpq)  # In-place assignment to the real part, using views
```

Comment:
@rbolgaryn I think you are overwriting F_real here. You might want to try `F_real[:] = `